AMW_angular/io/src/app/settings/deployment-parameter/deployment-parameter.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, ViewChild } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { ToastComponent } from '../../shared/elements/toast/toast.component';
+import { HttpClient } from '@angular/common/http';
+
+type Key = { key: string; value: any };
+
+@Component({
+  selector: 'app-deployment-parameter',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule, ToastComponent, FormsModule],
+  templateUrl: './deployment-parameter.component.html',
+  styleUrl: './deployment-parameter.component.scss',
+})
+export class DeploymentParameterComponent {
+  keyName = '';
+  paramKeys: Key[] = [];
+
+  @ViewChild(ToastComponent) toast: ToastComponent;
+
+  constructor(private http: HttpClient) {
+    http.get<Key[]>('/AMW_rest/resources/deployments/deploymentParameterKeys/').subscribe({
+      next: (data) => {
+        this.paramKeys = data;
+      },
+    });
+  }
+
+  addKey(): void {
+    const trimmedKeyName = this.keyName.trim();
+    if (trimmedKeyName.length > 0 && trimmedKeyName.toLowerCase() !== 'null') {
+      this.http
+        .post<Key>('/AMW_rest/resources/deployments/deploymentParameterKeys/', { key: this.keyName, value: null })
+        .subscribe({
+          next: (newKey) => {
+            this.paramKeys.push(newKey);
```

Comment:
The new entry will no be displayed. This is because of the response type which will not map to a Key-entity. You have to create a new Key-object first and then push it to the array.

(Hint: check the response value in the network tab)